src/helpers/serverModuleHelpers.js

```diff
@@ -0,0 +1,122 @@
+const { existsSync } = require('node:fs')
+const { readFile, writeFile, rename, rm } = require('node:fs/promises')
+const { parse, join } = require('node:path')
+
+const { satisfies } = require('semver')
+
+const getAngularJson = require('./getAngularJson')
+const { getProject } = require('./setUpEdgeFunction')
+
+// eslint-disable-next-line no-inline-comments
+const NetlifyServerTsCommonEngine = /* typescript */ `import { CommonEngine } from '@angular/ssr/node'
+import { render } from '@netlify/angular-runtime/common-engine'
+
+const commonEngine = new CommonEngine()
+
+export default async function HttpHandler(request: Request, context: any): Promise<Response> {
+  return await render(commonEngine)
+}
+`
+
+// eslint-disable-next-line no-inline-comments
+const NetlifyServerTsAppEngine = /* typescript */ `import { AngularAppEngine, createRequestHandler } from '@angular/ssr'
+const angularAppEngine = new AngularAppEngine()
+
+// @ts-expect-error - createRequestHandler expects a function with single Request argument and doesn't allow context argument
+export const reqHandler = createRequestHandler(async (request: Request, context: any) => {
+  const result = await angularAppEngine.handle(request, context)
+  return result || new Response('Not found', { status: 404 })
+})
+`
+
+let needSwapping = false
+let serverModuleLocation
+let serverModuleBackupLocation
+
+/**
+ * Inspect content of server module and determine which engine is used
+ * @param {string} serverModuleContents
+ * @returns {'AppEngine' | 'CommonEngine' | undefined}
+ */
+const getUsedEngine = function (serverModuleContents) {
```

Comment:
Yeah, this seems ok as a last-minute last-resort option, but it's beyond brittle and leaky and realistic scenarios that would break are easy to imagine (not purely hypothetical):

```js
// TODO Switch to AppEngine when stable
import { CommonEngine } from '@angular/ssr/node'
/* ... */
```

Reply:
With the things about known content ( https://github.com/netlify/angular-runtime/pull/216#discussion_r1844589743 ) I think what seems most reasonable is to do this:
 1. Compare against known default variants of `server.ts` file and if those match exactly - we can swap automatically as then we know there are no customizations there and our default is 1:1 replacement.
 2. If `server.ts` is unknown - we can still try to find those keywords - but if both `CommonEngine` and `AppEngine` keywords are found - like in example above - the result of checking is "not sure". But we wouldn't use this information (even if just CommonEngine or AppEngine is found) and instead we only use this information to generate meaningful error message - if single engine is used - error message just show content their `server.ts` should have to use given engine - if no engines or both are found - we display both variants in error message (or we just display both variants always and just mention engine we discovered as most likely what "user wants" to not hide any information - just attempt to prioritize one we ~think is most relevant to user setup.

The part that is not clear to me however is how exactly I could check for wether current `server.ts` file is fine for our edge handler usage (aka wether user already use module in Netlify compatible way) - for now this is done by checking if `express` keyword is used which has pretty much all the same problems as selecting engine that is being used (easy to produce false positives)